### DISCUSSION TITLE
Dev tools feature: button to pause/unpause all refetching

### DISCUSSION THREAD

**Discussion by mturley** [TanStack/query#2423] [Ideas]
I have a UI with `refetchInterval` set on all my queries, and when trying to debug some things I have console.logs that trigger on every render. It would be really nice to be able to wait for a few render cycles and then temporarily pause RQ so I don't get any new fetches/renders until I'm done looking through my log.

**Comment by TkDodo** [comment]
Not sure how to do that from just the devtools. Maybe you can do that in your code by setting the `enabled` option conditionally ?

**Reply by mturley** [reply]
Yeah, I was thinking just some kind of global switch in the dev tools that overrides `enabled` and sets all queries to `enabled: false` while it's turned on essentially. A toggle for "disable all".

**Reply by TkDodo** [reply]
the devtools just observe the queryCache, and then interact with it via the public methods of the queryClient, like `queryClient.invalidateQueries`, `queryClient.refetchQueries` etc. We could `setQueryDefaults` on the queries, but that doesn't guarantee that it hasn't been overwritten by actual `useQuery` calls.

**Reply by mturley** [reply]
ah that makes sense, okay. so the internals would need to be modified to make this possible, which it sounds like isn't worth it.
